ext/web/03_abort_signal.js

```diff
@@ -91,25 +178,36 @@ class AbortSignal extends EventTarget {
 
     const event = new Event("abort");
     setIsTrusted(event, true);
-    this.dispatchEvent(event);
+    super.dispatchEvent(event);
     if (algos !== null) {
       for (const algorithm of new SafeSetIterator(algos)) {
         algorithm();
       }
     }
+
+    if (this[dependentSignals] !== null) {
+      const dependentSignalArray = this[dependentSignals].toArray();
+      for (let i = 0; i < dependentSignalArray.length; ++i) {
+        const dependentSignal = dependentSignalArray[i];
+        dependentSignal[signalAbort](reason);
+      }
+    }
   }
 
   [remove](algorithm) {
     this[abortAlgos] && SetPrototypeDelete(this[abortAlgos], algorithm);
   }
 
   constructor(key = null) {
-    if (key != illegalConstructorKey) {
+    if (key !== illegalConstructorKey) {
       throw new TypeError("Illegal constructor.");
     }
     super();
     this[abortReason] = undefined;
     this[abortAlgos] = null;
```

Comment:
I think this is currently supposed to be an empty set on construction, see https://dom.spec.whatwg.org/#abortsignal-abort-algorithms. That should also cascade on a bunch of "null-check plus re-initialization" which may be redundant now.

Reply:
This was done for performance reasons in #12165, though not sure how ideal this is. I wouldn't worry about this though as it works as expected.